metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/utils/EntityMergeEngine.java

```diff
@@ -373,26 +390,85 @@ private static void convertAndAddEntity(RDF rdf, EnrichmentBase enrichmentBase,
     } else if (enrichmentBase instanceof Timespan) {
       entity = convertAndAddEntity((Timespan) enrichmentBase, EntityMergeEngine::convertTimeSpan,
           rdf::getTimeSpanList, rdf::setTimeSpanList);
+    } else if (enrichmentBase instanceof Organization) {
+      entity = convertAndAddEntity((Organization) enrichmentBase,
+          EntityMergeEngine::convertOrganization, rdf::getOrganizationList,
+          rdf::setOrganizationList);
     } else {
       throw new IllegalArgumentException("Unknown entity type: " + enrichmentBase.getClass());
     }
 
-    // Append it to the europeana proxy if needed, regardless of whether entity is new or existing.
-    if (!CollectionUtils.isEmpty(proxyLinkTypes)) {
-      RdfProxyUtils.appendLinkToEuropeanaProxy(rdf, entity.getAbout(), proxyLinkTypes);
+    return entity;
+  }
+
+  /**
+   * Merge entities in a record.
+   *
+   * @param rdf The RDF to enrich
+   * @param enrichmentBaseList The information to append
+   * @param searchTermContext the search term context
+   */
+  public void mergeSearchEntities(RDF rdf, List<EnrichmentBase> enrichmentBaseList,
+      SearchTermContext searchTermContext) {
+    for (EnrichmentBase base : enrichmentBaseList) {
+      final AboutType aboutType = convertAndAddEntity(rdf, base);
+      if (isProxyFieldType(searchTermContext.getFieldTypes())) {
+        appendLinkToEuropeanaProxy(rdf, aboutType.getAbout(),
+            searchTermContext.getFieldTypes().stream().map(ProxyFieldType.class::cast)
+                .collect(Collectors.toSet()));
+      } else {
+        //Replace matching values in Aggregation
+        replaceValueWithLinkInAggregation(rdf, aboutType.getAbout(), searchTermContext);
+      }
     }
   }
 
   /**
    * Merge entities in a record.
+   * <p>This method is when enrichment is performed where we <b>do</b> want to add the generated
+   * links to the europeana proxy</p>
    *
    * @param rdf The RDF to enrich
    * @param enrichmentBaseList The information to append
+   * @param referenceTermContext the reference term context
    */
-  public void mergeEntities(RDF rdf, List<EnrichmentBase> enrichmentBaseList,
-      Set<FieldType> proxyLinkTypes) {
+  public void mergeReferenceEntities(RDF rdf, List<EnrichmentBase> enrichmentBaseList,
+      ReferenceTermContext referenceTermContext) {
     for (EnrichmentBase base : enrichmentBaseList) {
-      convertAndAddEntity(rdf, base, proxyLinkTypes);
+      final AboutType aboutType = convertAndAddEntity(rdf, base);
+      if (referenceTermContext != null) {
+        appendLinkToEuropeanaProxy(rdf, aboutType.getAbout(),
+            referenceTermContext.getFieldTypes().stream().map(ProxyFieldType.class::cast)
+                .collect(Collectors.toSet()));
+      }
+    }
```

Comment:
This is almost the same as the method above. I think you can call that one from here.

Reply:
Not sure exactly what you mean and also it might become a bit too intertwined.
I'm leaving this as is.

Reply:
OK. Perhaps this is something for the future.